kafka-plugins-client/src/main/java/io/cdap/plugin/kafka/sink/KafkaBatchSink.java

```diff
@@ -118,7 +118,14 @@ public void transform(StructuredRecord input, Emitter<KeyValue<Text, Text>> emit
         emitter.emit(new KeyValue<>((Text) null, new Text(body)));
       } else {
         String key = input.get(producerConfig.key);
-        emitter.emit(new KeyValue<>(new Text(key), new Text(body)));
+        // The key inside the message may be null if it is not specified while publishing the event.
+        // https://stackoverflow.com/questions/29511521
+        if (key != null) {
+          emitter.emit(new KeyValue<>(new Text(key), new Text(body)));
+        } else {
+          throw new RuntimeException(String.format("Key for the event: %s published into Kafka must exist in the " +
```

Comment:
nit: Change the message to indicate that the key field value is null.

Reply:
Can a unit test be added for this ?

Reply:
> Can a unit test be added for this ?

I'm working on adding a unit test for the `null` key use case

Reply:
Unit test added 399af3d6b9ad9fc47f2234d2790018cfd584144f